scripts/vendor-github-repo.ts

```diff
@@ -1,28 +1,61 @@
-import { ChildProcess, execFile } from 'child_process';
+import { execa, ResultPromise } from 'execa';
 import * as fs from 'fs/promises';
 import * as tmp from 'tmp';
 import * as yargs from 'yargs';
 
-function isErrorExists(e: NodeJS.ErrnoException): boolean {
-  return e.code === 'EEXIST';
-}
-
 type GithubSource = {
   org: string;
   repo: string;
 } & ({ tag: string } | { ref: string });
 
-async function wait(p: ChildProcess): Promise<void> {
-  p.stderr?.pipe(process.stderr);
-  p.stdout?.pipe(process.stdout);
-  return new Promise((resolve, reject) => {
-    p.on('exit', (code) => {
-      if (code === 0) {
-        resolve();
-      } else {
-        reject(new Error(`Process exited with code ${code}`));
-      }
-    });
+export function getDescription(source: GithubSource): string {
+  if ('tag' in source) {
+    return `${source.org}/${source.repo}#${source.tag}`;
+  } else if ('ref' in source) {
+    return `${source.org}/${source.repo}#${source.ref}`;
+  } else {
+    throw new Error('Invalid GithubSource, must have either tag or ref');
+  }
```

Comment:
nit: can be simplified to avoid if/else if structure.
```suggestion
  const tagOrRef = source.tag ?? source.ref;

  if (tagOrRef) {
    return `${source.org}/${source.repo}#${tagOrRef}`;
  }

  throw new Error('Invalid GithubSource, must have either tag or ref');
```

Reply:
the type signature only allows either `tag` or `ref`, not both properties, so this won't work